nomad/leader.go

```diff
@@ -149,6 +149,34 @@ func (s *Server) monitorLeadership() {
 	}
 }
 
+func (s *Server) leadershipTransferToServer(id raft.ServerID, addr raft.ServerAddress) error {
+	if lAddr, lID := s.raft.LeaderWithID(); id == lID && addr == lAddr {
+		s.logger.Debug("leadership transfer to current leader is a no-op")
+		return nil
+	}
+	retryCount := 3
+	for i := 0; i < retryCount; i++ {
```

Comment:
Just out of curiosity, what are the circumstances in which this will need to retry, and can safely do so such that we wouldn't want to immediately return to the user?

Reply:
Resolved!